File: engine/model/src/main/scala/tools/packet/character/interaction/PlayerShopOwnerUpdate.scala
package tools.packet.character.interaction

import net.opcodes.SendOpcode
import server.maps.MaplePlayerShopSoldItem
import tools.packet.PacketInput

class PlayerShopOwnerUpdate(private var _soldItem: MaplePlayerShopSoldItem, private var _position: Int) extends PacketInput {
  def soldItem: MaplePlayerShopSoldItem = _soldItem

  def position: Int = _position

  override def opcode(): SendOpcode = SendOpcode.PLAYER_INTERACTION
}

File: engine/engine-model/src/main/java/tools/packet/character/interaction/PlayerShopOwnerUpdate.java
package tools.packet.character.interaction;

import net.opcodes.SendOpcode;
import server.maps.MaplePlayerShopSoldItem;
import tools.packet.PacketInput;

public record PlayerShopOwnerUpdate(MaplePlayerShopSoldItem soldItem, Integer position) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PLAYER_INTERACTION;
   }
}